### ISSUE TITLE
General exception error has occurred in the optimizer in Informix code generation, when generating only one schema and <includeSystemIndexes/> is turned on

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#10370]
There seems to be a bug in Informix for the following query produced by the code generator on IBM Informix Dynamic Server Version 14.10.FC3DE

```sql
select
  trim(informix.systables.owner) owner,
  trim(informix.systables.tabname) tabname,
  trim(informix.sysindexes.idxname) idxname,
  trim(informix.sysindexes.idxtype) idxtype,
  informix.sysindexes.part1,
  informix.sysindexes.part2,
  informix.sysindexes.part3,
  informix.sysindexes.part4,
  informix.sysindexes.part5,
  informix.sysindexes.part6,
  informix.sysindexes.part7,
  informix.sysindexes.part8,
  informix.sysindexes.part9,
  informix.sysindexes.part10,
  informix.sysindexes.part11,
  informix.sysindexes.part12,
  informix.sysindexes.part13,
  informix.sysindexes.part14,
  informix.sysindexes.part15,
  informix.sysindexes.part16
from informix.sysindexes
  join informix.systables
    on informix.sysindexes.tabid = informix.systables.tabid
where (
  informix.sysindexes.owner in ('informix')
  and not (exists (
    select
      alias_1.alias_1_0,
      alias_1.alias_1_1,
      alias_1.alias_1_2
    from (
      select
        informix.sysconstraints.tabid,
        informix.sysconstraints.owner,
        informix.sysconstraints.idxname
      from informix.sysconstraints
    ) alias_1 (alias_1_0, alias_1_1, alias_1_2)
    where row (informix.systables.tabid, informix.sysindexes.owner, informix.sysindexes.idxname) 
        = row (alias_1.alias_1_0, alias_1.alias_1_1, alias_1.alias_1_2)
  ))
)
order by
  informix.systables.owner asc,
  informix.systables.tabname asc,
  informix.sysindexes.idxname asc
```

The problem seems to be related to the `in` predicate with a single value and the `not exists()` predicate with the `row(...) = row(...)` predicate, which is an emulation of `row(...).notIn(select)`

A workaround would be to expand that `row` predicate:

```sql
    where informix.systables.tabid = alias_1.alias_1_0
    and informix.sysindexes.owner = alias_1.alias_1_1
    and informix.sysindexes.idxname = alias_1.alias_1_2
```

Or to add a dummy value in the `in` predicate:

```sql
  informix.sysindexes.owner in ('informix', null)
```

**Comment by lukaseder** [comment]
This only happens when `<includeSystemIndexes/>` is turned on, otherwise, the subquery is not generated

**Comment by lukaseder** [comment]
We'll force the emulation of 

```sql
    where row (informix.systables.tabid, informix.sysindexes.owner, informix.sysindexes.idxname) 
      = row (alias_1.alias_1_0, alias_1.alias_1_1, alias_1.alias_1_2)
```

```sql
    where informix.systables.tabid = alias_1.alias_1_0
    and informix.sysindexes.owner = alias_1.alias_1_1
    and informix.sysindexes.idxname = alias_1.alias_1_2
```

This problem might appear elsewhere in user code, not just in the code generator.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.14.0. Backport scheduled for 3.13.3 (#10371)
